lib/auth/clt.go

```diff
@@ -1534,6 +1534,11 @@ type IdentityService interface {
 	// CreatePrivilegeToken creates a privilege token for the logged in user who has successfully re-authenticated with their second factor.
 	// A privilege token allows users to perform privileged action eg: add/delete their MFA device.
 	CreatePrivilegeToken(ctx context.Context, req *proto.CreatePrivilegeTokenRequest) (*types.UserTokenV3, error)
+
+	// UpdateHeadlessAuthenticationState updates a headless authentication state.
```

Comment:
Can you mention what happens when the request doesn't exist? Is the error returned, or is the request created? (ignore this if you think that calling the method `Update` instead of `Upsert` is enough).